### DISCUSSION TITLE
run tests spec file

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#16926] [Q&A Tools] [answered]
What's the difference between the `-i` and `--spec-file` flags for `run_tests`? They both mention running a specific test spec file, but I can't get the `-i` switch to do anything.

**Comment by aeslaughter** [comment, marked as answer]
`-i` sets the input file name (defaults to "tests"), https://github.com/idaholab/moose/blob/25b11926d6dc2fe60fb9ebd63bdcd564d081ace9/python/TestHarness/TestHarness.py#L932

`--spec-file` seems to be for setting up the search directory or defining a specific file to execute. https://github.com/idaholab/moose/blob/25b11926d6dc2fe60fb9ebd63bdcd564d081ace9/python/TestHarness/TestHarness.py#L222

**Reply by jessecarterMOOSE** [reply]
Ah, I see now. Thanks.

**Reply by jessecarterMOOSE** [reply]
So if for some reason I called my `tests` files `mytests`, I would run `./run_tests -i mytests`, but if I were to run a specific set of tests in a file it would be something like `./run_tests tests/kernel/simple_diffusion/tests`.

I had thought it was the opposite because `-i` for a moose executable means "run this specific input file".

**Reply by aeslaughter** [reply]
These are some terrible names, but you are one of only a few that have ever used a different tests file name.

**Reply by jessecarterMOOSE** [reply]
Ha, no, that was just a hypothetical to help me understand.
